pandas/tests/indexing/test_at.py

```diff
@@ -47,6 +37,68 @@ def test_selection_methods_of_assigned_col():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "value_to_assign",
+    [
+        "a",
+        1,
+        1.0,
+        np.nan,
+        True,
+        ("a",),
+        ["a"],
+        datetime(2000, 1, 2, tzinfo=timezone.utc),
+    ],
+)
+def test_at_setitem_expansion__df(value_to_assign):
+    # GH33544
+    # GH30649
+    result = DataFrame({0: [1, 2]}, index=[3, 2])
+
+    result.at[1, 1] = value_to_assign
+    expected_result = DataFrame(
+        {0: [1, 2, np.nan], 1: [np.nan, np.nan, value_to_assign]}, index=[3, 2, 1]
+    )
+    if isinstance(value_to_assign, datetime):
+        expected_result = expected_result.astype({1: object})
+
+    tm.assert_frame_equal(result, expected_result)
+
+
+@pytest.mark.parametrize(
+    "value_to_assign",
+    [
+        "a",
+        1,
+        1.0,
+        np.nan,
+        True,
+        ("a",),
+        ["a"],
+        {"a": "b"},
+        datetime(2000, 1, 2, tzinfo=timezone.utc),
+        np.array([1]),
+        Series([1]),
+    ],
+)
+@pytest.mark.filterwarnings("ignore::FutureWarning")
```

Comment:
what is showing the warning?

Reply:
The following warnings are present (FutureWarning was here earlier; UserWarning occurred just now, not sure if this is because of my setup or something else changed in the meantime?):
- FutureWarning: `Behavior when concatenating bool-dtype and numeric-dtype arrays is deprecated; in a future version these will cast to object dtype (instead of coercing bools to numeric values). To retain the old behavior, explicitly cast bool-dtype arrays to numeric dtype.` (related to the TODO comment)
- UserWarning: `pyarrow requires pandas 0.23.0 or above, pandas 0+unknown is installed. Therefore, pandas-specific integration is not used.`